core/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2Authorization.java

```diff
@@ -16,16 +16,182 @@
 package org.springframework.security.oauth2.server.authorization;
 
 import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
 
+import java.util.Collections;
 import java.util.Map;
+import java.util.Objects;
 
 /**
+ * Represents a collection of attributes which describe an OAuth 2.0 authorization context.
+ *
  * @author Joe Grandja
+ * @author Krisztian Toth
  */
 public class OAuth2Authorization {
 	private String registeredClientId;
 	private String principalName;
 	private OAuth2AccessToken accessToken;
 	private Map<String, Object> attributes;
 
+	private OAuth2Authorization() {
+	}
+
+	public String getRegisteredClientId() {
+		return registeredClientId;
+	}
+
+	public String getPrincipalName() {
+		return principalName;
+	}
+
+	public OAuth2AccessToken getAccessToken() {
+		return accessToken;
+	}
+
+	public Map<String, Object> getAttributes() {
+		return attributes;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		OAuth2Authorization that = (OAuth2Authorization) o;
+		return Objects.equals(registeredClientId, that.registeredClientId) &&
+				Objects.equals(principalName, that.principalName) &&
+				Objects.equals(accessToken, that.accessToken) &&
+				Objects.equals(attributes, that.attributes);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(registeredClientId, principalName, accessToken, attributes);
+	}
+
+	@Override
+	public String toString() {
```

Comment:
Please remove `toString()` as there will be sensitive information here and we don't want it displayed